src/neo/SmartContract/Native/Oracle/OracleContract.cs

```diff
@@ -0,0 +1,429 @@
+                        return GetHttpConfig(engine.Snapshot).ToStackItem(engine.ReferenceCounter);
+                    }
+            }
+
+            return false;
+        }
+
+        /// <summary>
+        /// Get HttpConfig
+        /// </summary>
+        /// <param name="snapshot">snapshot</param>
+        /// <returns>value</returns>
+        public HttpConfig GetHttpConfig(StoreView snapshot)
+        {
+            var storage = snapshot.Storages.GetAndChange(CreateStorageKey(Prefix_Config, Encoding.UTF8.GetBytes(HttpConfig.Key)));
+            return storage.GetInteroperable<HttpConfig>();
+        }
+
+        /// <summary>
+        /// Set PerRequestFee
+        /// </summary>
+        /// <param name="engine">VM</param>
+        /// <param name="args">Parameter Array</param>
+        /// <returns>Returns true if the execution is successful, otherwise returns false</returns>
+        [ContractMethod(0_03000000, ContractParameterType.Boolean, CallFlags.AllowModifyStates, ParameterTypes = new[] { ContractParameterType.Integer }, ParameterNames = new[] { "fee" })]
+        private StackItem SetPerRequestFee(ApplicationEngine engine, Array args)
+        {
+            StoreView snapshot = engine.Snapshot;
+            UInt160 account = GetOracleMultiSigAddress(snapshot);
+            if (!InteropService.Runtime.CheckWitnessInternal(engine, account)) return false;
+            int perRequestFee = (int)args[0].GetBigInteger();
+            if (perRequestFee <= 0) return false;
+            StorageItem storage = snapshot.Storages.GetAndChange(CreateStorageKey(Prefix_PerRequestFee));
+            storage.Value = BitConverter.GetBytes(perRequestFee);
+            return true;
+        }
+
+        /// <summary>
+        /// Get PerRequestFee
+        /// </summary>
+        /// <param name="engine">VM</param>
+        /// <param name="args">Parameter Array</param>
+        /// <returns>Value</returns>
+        [ContractMethod(0_01000000, ContractParameterType.Integer, requiredCallFlags: CallFlags.AllowStates)]
+        private StackItem GetPerRequestFee(ApplicationEngine engine, Array args)
+        {
+            return new Integer(GetPerRequestFee(engine.Snapshot));
+        }
+
+        /// <summary>
+        /// Get PerRequestFee
+        /// </summary>
+        /// <param name="snapshot">snapshot</param>
+        /// <returns>Value</returns>
+        public int GetPerRequestFee(StoreView snapshot)
+        {
+            StorageItem storage = snapshot.Storages.TryGet(CreateStorageKey(Prefix_PerRequestFee));
+            if (storage is null) return 0;
+            return BitConverter.ToInt32(storage.Value);
+        }
+
+        /// <summary>
+        /// Oracle get the hash of the current OracleFlow [Request/Response]
+        /// </summary>
+        [ContractMethod(0_01000000, ContractParameterType.Boolean, requiredCallFlags: CallFlags.None)]
+        private StackItem GetHash(ApplicationEngine engine, Array args)
```

Comment:
Maybe we need a better name. It conflicts with hash of oracle contract.

Reply:
Done